src/crypto-api/CryptoEvent.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2024 The Matrix.org Foundation C.I.C.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export enum CryptoEvent {
+    /**
+     * Fires when the trust status of a user changes.
+     * @param {string} userId - the user whose trust status changed
+     * @param {UserVerificationStatus} userTrustLevel - the new trust level
+     */
+    UserTrustStatusChanged = "userTrustStatusChanged",
+    /**
+     * Fires when the key backup status changes.
+     * @param {boolean} status - true if the key backup is enabled, false otherwise.
+     */
+    KeyBackupStatus = "crypto.keyBackupStatus",
+    /**
+     * Fires when we failed to back up the keys
+     * @param {string} errorCode - the error code of the error that occurred
+     */
+    KeyBackupFailed = "crypto.keyBackupFailed",
+    /**
+     * Fires when the number of sessions that can be backed up changes.
+     * @param {number} remaining - the remaining number of sessions that can be backed up.
+     */
+    KeyBackupSessionsRemaining = "crypto.keyBackupSessionsRemaining",
+    /**
+     * Fires when a new valid backup decryption key is in cache.
+     * This will happen when a secret is received from another session, from secret storage,
+     * or when a new backup is created from this session.
+     *
+     * The payload is the version of the backup for which we have the key for.
+     *
+     * This event is only fired by the rust crypto backend.
+     */
+    KeyBackupDecryptionKeyCached = "crypto.keyBackupDecryptionKeyCached",
+    /**
+     * Fires when a key verification request is received.
+     * @param {VerificationRequest} - the request that was received
+     */
+    VerificationRequestReceived = "crypto.verificationRequestReceived",
+    /**
+     * Fires whenever the stored devices for a user will be updated
+     * @deprecated Use {@link DevicesUpdated} instead
+     * @param {string[]} users - A list of user IDs that will be updated
+     * @param {boolean} initialFetch - If true, the store is empty (apart
+     *     from our own device) and is being seeded.
+     */
+    WillUpdateDevices = "crypto.willUpdateDevices",
```

Comment:
I had to keep it to avoid circular import between the old crypto and the crypto api.
The other solution is to remove the event usage in the rust-crypto and the crypto api but it will be breaking change.